### ISSUE TITLE
Tests fail on 32 bit linux

### ISSUE THREAD

**Issue by seanlth** [pornel/lodepng-rust#11]
Tests are failing on debian 32 bit using Rust 1.7

From what I can tell the issue seems to crop up when calling into C.

**Comment by pornel** [comment]
Sorry, I couldn't find a 32-bit machine, or even a working VM, to test it.

I've tested it on 64-bit Linux, and it passes tests there.

It's possible that there's some mismatch between C `int`/`long` and Rust sizes that causes problems.

**Comment by petevine** [comment]
Once patched, all tests pass on ARM presently.

**Comment by ideasman42** [comment]
Confirmed with 32bit Arch Linux, 
output from valgrind may help: [valgrind.txt](https://github.com/pornel/lodepng-rust/files/507735/valgrind.txt)

Rebuilt with debug cflags:
```
diff --git a/Makefile b/Makefile
index 49c6d69..f708209 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS ?= -O3 -fPIC
+CFLAGS ?= -O0 -ggdb3 -fno-inline -fPIC
 OUT_DIR ?= .
 VENDOR_DIR ?= vendor
 LIB = $(OUT_DIR)/liblodepng.a
```

**Comment by pornel** [comment]
Likely fixed in 2.0
